app/qml/FeaturePanel.qml

```diff
@@ -134,10 +142,36 @@ Drawer {
             height: InputStyle.rowHeightHeader
             width: parent.width
             color: InputStyle.clrPanelMain
+            fontBtnColor: InputStyle.highlightColor
             rowHeight: InputStyle.rowHeightHeader
             titleText: featurePanel.formState === "Edit" ? qsTr("Edit Feature") : qsTr("Feature")
+            backIconVisible: false
+            backText: qsTr("Cancel")
+
+            onBack: {
+              featurePanel.close()
+            }
+
+            Text {
+                id: saveButtonText
+                text: qsTr("Save")
+                enabled: featureForm.model.constraintsHardValid
+                color: enabled ? InputStyle.highlightColor : "red"
+                font.pixelSize: InputStyle.fontPixelSizeNormal
+                height: header.rowHeight
+                verticalAlignment: Text.AlignVCenter
+                horizontalAlignment: Text.AlignLeft
+                anchors.right: parent.right
+                anchors.bottom: parent.bottom
+                anchors.top: parent.top
+                anchors.rightMargin: InputStyle.panelMargin // same as back button
+
+                MouseArea {
+                  anchors.fill: parent
+                  onClicked: featureForm.save()
+                }
+            }
```

Comment:
I guess the save button should only be visible when feature form is in edit mode?

```qml
        visible: featureForm.state === "edit"
```